### PR TITLE
Test target support

### LINKED ISSUE #1: Support specifying dependencies for Test targets

### PR DESCRIPTION
**Pull request by fredpi**
Closes #1 #6.

### CODE REVIEW

File: Sources/AccioKit/Services/XcodeProjectIntegrationService.swift [hunk 1]
```diff
@@ -16,6 +16,28 @@ final class XcodeProjectIntegrationService {
...
+
+        let targetNames = targetsToKeep.map { $0.targetName }
+        let groupsToRemove = dependenciesGroup.children.filter { !targetNames.contains($0.name ?? "") }
+
+        guard !groupsToRemove.isEmpty else { return }
+
+        print("Removing unnecessary groups \(groupsToRemove.compactMap { $0.name }) from group 'Dependencies' ...", level: .info)
+
+        for groupToRemove in groupsToRemove {
+            dependenciesGroup.children.removeAll { $0 == groupToRemove }
```
Comment (Jeehut) [inline Sources/AccioKit/Services/XcodeProjectIntegrationService.swift]: Here, you should load the franeworksbuildPhase of the target named like the group (if existing) and remove all frameworks within the group from the frameworks build phase.

**Review by Jeehut — CHANGES_REQUESTED** [review]
Looks good overall, just missing the cleanup of frameworks from build phase at the commented place.

File: README.md [hunk 2]
```diff
@@ -131,7 +131,7 @@ Accio uses the [official SwiftPM manifest format](https://github.com/apple/swift
 Here's an example `Package.swift` file with multiple dependencies specified:
 
 ```swift
-// swift-tools-version:4.2
+// swift-tools-version:5.0
```
Comment (Jeehut) [inline README.md:134]: Could you also add this change to the `Package.swift` file created by the `init` command?

File: Sources/AccioKit/Services/XcodeProjectIntegrationService.swift [hunk 3]
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
         self.workingDirectory = workingDirectory
     }
 
-    func updateDependencies(of appTarget: AppTarget, for platform: Platform, with frameworkProducts: [FrameworkProduct]) throws {
-        print("Relinking build products with targets in Xcode project ...", level: .info)
+    func unlinkAndRemoveGroupsOfUnnededTargets(keepingTargets targetsToKeep: [AppTarget]) throws {
+        guard let projectName = targetsToKeep.first?.projectName else { return }
+
+        // find groups of which the associated target is not be kept
```
Comment (Jeehut) [inline Sources/AccioKit/Services/XcodeProjectIntegrationService.swift]: "is not be kept"? Are you missing a "to" there?

File: Sources/AccioKit/Services/XcodeProjectIntegrationService.swift [hunk 4]
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
...
+        for groupToRemove in groupsToRemove {
+            guard !groupToRemove.children.isEmpty, let groupName = groupToRemove.name else { continue }
+
+            let frameworkBuildPhase = pbxproj.targets(named: groupName).compactMap { $0.buildPhases.first(where: { $0.buildPhase == .frameworks }) as? PBXFrameworksBuildPhase }.first
+
+            let printSuffix = frameworkBuildPhase == nil ? "..." : "& unlinking from target '\(groupName)' ..."
+            print("Removing frameworks \(groupToRemove.children.compactMap { $0.name }) from project navigator group '\(groupName)' \(printSuffix)", level: .info)
+
+            groupToRemove.children.removeAll()
+            frameworkBuildPhase?.files.removeAll { file in groupToRemove.children.compactMap { $0.name }.contains { $0 == file.file?.name } }
```
Comment (Jeehut) [inline Sources/AccioKit/Services/XcodeProjectIntegrationService.swift]: How is `groupToRemove.children` here not empty after calling `removeAll` right one line before this?

File: Sources/AccioKit/Services/XcodeProjectIntegrationService.swift [hunk 5]
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
...
+        for groupToRemove in groupsToRemove {
+            dependenciesGroup.children.removeAll { $0 == groupToRemove }
+        }
+
+        // write project file
+        try projectFile.write(path: Path(xcodeProjectPath), override: true)
+    }
+
+    func clearDependenciesFolder() throws {
+        let dependenciesPath = "\(workingDirectory)/\(Constants.dependenciesPath)/*"
```
Comment (Jeehut) [inline Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:83]: Why the `/*` suffix? That's not needed afaik when running `rm -rf`.

File: Sources/AccioKit/Services/XcodeProjectIntegrationService.swift [hunk 6]
```diff
@@ -16,15 +16,58 @@ final class XcodeProjectIntegrationService {
...
+        try bash("rm -rf \(dependenciesPath)")
+    }
+
+    func updateDependencies(of appTarget: AppTarget, for platform: Platform, with frameworkProducts: [FrameworkProduct]) throws {
         let dependenciesPlatformPath = "\(workingDirectory)/\(Constants.dependenciesPath)/\(platform.rawValue)"
-        let copiedFrameworkProducts: [FrameworkProduct] = try copyFrameworkProducts(frameworkProducts, to: dependenciesPlatformPath)
-        try linkFrameworks(copiedFrameworkProducts, with: appTarget, for: platform)
+        let copiedFrameworkProducts: [FrameworkProduct] = try copy(frameworkProducts: frameworkProducts, of: appTarget, to: dependenciesPlatformPath)
```
Comment (Jeehut) [inline Sources/AccioKit/Services/XcodeProjectIntegrationService.swift:89]: Nice catch there. 👍

**Review by Jeehut — COMMENTED** [review]
Great work overall, please fix the commented lines and after adding the copy frameworks phase in test targets, it should be good to merge! 👍

File: Sources/AccioKit/Services/XcodeProjectIntegrationService.swift (hunk 4 above)
Comment (fredpi) [reply Sources/AccioKit/Services/XcodeProjectIntegrationService.swift]: Oh my bad 🤦‍♂️

**Review by Jeehut — APPROVED** [review]

**Comment by Jeehut** [comment]
Thank you for this, looks good, merging.

**Pull request merged**